### PR TITLE
Improve LSRA and other dump output

### PR DESCRIPTION
**Pull request by BruceForstall**
E.g.,

Update LSRA "Allocating Registers" table description. Now:
```
The following table has one or more rows for each RefPosition that is handled during allocation.
The columns are: (1) Loc: LSRA location, (2) RP#: RefPosition number, (3) Name, (4) Type (e.g. Def, Use,
Fixd, Parm, DDef (Dummy Def), ExpU (Exposed Use), Kill) followed by a '*' if it is a last use, and a 'D'
if it is delayRegFree, (5) Action taken during allocation. Some actions include (a) Alloc a new register,
(b) Keep an existing register, (c) Spill a register, (d) ReLod (Reload) a register. If an ALL-CAPS name
such as COVRS is displayed, it is a score name from lsra_score.h, with a trailing '(A)' indicating alloc,
'(C)' indicating copy, and '(R)' indicating re-use. See dumpLsraAllocationEvent() for details.
The subsequent columns show the Interval occupying each register, if any, followed by 'a' if it is
active, 'p' if it is a large vector that has been partially spilled, and 'i' if it is inactive.
Columns are only printed up to the last modified register, which may increase during allocation,
in which case additional columns will appear. Registers which are not marked modified have ---- in
their column.
```

Dump nodes added during resolution, e.g.:
```
   BB29 bottom (BB08->BB08): move V25 from STK to rdi (Critical)
N001 (  1,  1) [001174] ----------z                 t1174 =    LCL_VAR   int    V25 cse4          rdi REG rdi
```

Dump more data in the LSRA block sequence data:
```
-BB03( 16   )
-BB04(  4   )
+BB03 ( 16   ) critical-in critical-out
+BB04 (  4   ) critical-out
```

When dumping various flow bitvectors, annotate the bitvectors better:
```
-BB25 in gen out
-0000000000000000
-0000000000000003 CSE #01.c
-0000000000000003 CSE #01.c
+BB25
+ in: 0000000000000000
+gen: 0000000000000003 CSE #01.c
+out: 0000000000000003 CSE #01.c
```

Dump hoisting bitvectors using the function which sorts on local number:
```
-  USEDEF  (5)={V04 V00 V01 V02 V03}
+  USEDEF  (5)={V00 V01 V02 V03 V04}
```

Also, fix various typos and formatting.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@kunalspathak @dotnet/jit-contrib PTAL

File: src/coreclr/jit/lsra.cpp [hunk 1]
```diff
@@ -949,22 +949,33 @@ void LinearScan::setBlockSequence()
...
+    for (BasicBlock* block = startBlockSequence(); block != nullptr; block = moveToNextBlock())
     {
         JITDUMP(FMT_BB, block->bbNum);
-        JITDUMP("(%6s) ", refCntWtd2str(block->getBBWeight(compiler)));
 
-        if (blockInfo[block->bbNum].hasEHBoundaryIn)
+        const LsraBlockInfo& bi = blockInfo[block->bbNum];
+
+        // Note that predBBNum isn't set yet.
+        JITDUMP(" (%6s)", refCntWtd2str(bi.weight));
```
Comment (kunalspathak) [inline src/coreclr/jit/lsra.cpp:960]: Remind me, but `block->getBBWeight() == bi.weight` and hence this change is same as what he had until now?

File: src/coreclr/jit/lsra.cpp [hunk 2]
```diff
@@ -7572,13 +7596,21 @@ void LinearScan::addResolution(
...
     }
 
-    // We should never add resolution move inside BBCallAlwaysPairTail.
-    noway_assert(!block->isBBCallAlwaysPairTail());
-
+    assert(fromBlock != nullptr);
+    JITDUMP("   " FMT_BB " %s", block->bbNum, insertionPointString);
+    if (toBlock == nullptr)
+    {
+        // SharedCritical resolution has no `toBlock`.
```
Comment (kunalspathak) [inline src/coreclr/jit/lsra.cpp:7603]: May be helpful to print about `SharedCritical` and `fromBlock` at least?

**Review by kunalspathak — APPROVED** [review]
Thanks for doing this. Added some minor questions/suggestions.

File: src/coreclr/jit/gentree.h [hunk 3]
```diff
@@ -1194,7 +1194,7 @@ struct GenTree
         return (gtOper == GT_LCL_FLD || gtOper == GT_LCL_FLD_ADDR || gtOper == GT_STORE_LCL_FLD);
     }
 
-    inline bool OperIsLocalField() const
```
[the remark this answers is not in the document]
Comment (BruceForstall) [reply src/coreclr/jit/gentree.h:1197]: `inline` is implicit for function definitions within the body of a class declaration, so this is just to be consistent with all other cases that don't use it.

File: src/coreclr/jit/optcse.cpp [hunk 4]
```diff
@@ -1820,66 +1818,64 @@ class CSE_Heuristic
...
-                break; // early out,  we don't need to keep increasing frameSize
+                break; // early out, we don't need to keep increasing frameSize
             }
 #elif defined(TARGET_ARM)
             if (frameSize > 0x0400)
             {
                 // We likely have a large stack frame.
                 //
                 // Thus we might need to use large displacements when loading or storing
-                // to CSE LclVars that are not enregistered
```
[the remark this answers is not in the document]
Comment (BruceForstall) [reply src/coreclr/jit/optcse.cpp:1840]: Just reading code.

File: src/coreclr/jit/lsra.cpp (hunk 1 above)
Comment (BruceForstall) [reply src/coreclr/jit/lsra.cpp:960]: Exactly: no change to the output

**Pull request merged**